File: 0088-merge-sorted-array-合并两个有序数组/0088.cpp
class Solution {
private:
    inline void insert(vector<int>& A, int index, int value) {
        for(int i = A.size() - 1; i > index; --i) {
            A[i] = A[i-1];
        }
        A[index] = value;
    }
public:
    void merge(vector<int>& A, int m, vector<int>& B, int n) {
        int j = 0;
        for(int i = 0; i < B.size(); ++i) {
            while(true) {
                if(A[j] > B[i] || j >= m + i) {
                    insert(A, j, B[i]);
                    break;
                }
                ++j;
            }
        }
    }
};

File: 0088-merge-sorted-array-合并两个有序数组/0088.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        p = 0
        p1 = m - 1
        p2 = n - 1
        while p2 >= 0:
            p -= 1
            if p1 < 0:
                nums1[p] = nums2[p2]
                p2 -= 1
            elif nums2[p2] >= nums1[p1]:
                nums1[p] = nums2[p2]
                p2 -= 1
            else:
                nums1[p] = nums1[p1]
                p1 -= 1